odoo/api.py

```diff
@@ -839,6 +839,21 @@ def __init__(self, registry):
         self.tocompute = defaultdict(OrderedSet)
         # pending updates {model: {id: {field: value}}}
         self.towrite = defaultdict(lambda: defaultdict(dict))
+        # {model: {record_id: OrderedSet(key_t_cache)}}
+        # It use to rollback t-cache pollution due to a write after a t-cache set value on the shared memory
+        # Example:
+        # The record_1 of model A is used in the t-cache key of template X, in the same transaction:
+        # - write on record_1 : write_date (new) change
+        # - render template X : the render result is put in cache with key C containing the write_date (new) of record_1
+        # - rewrite on record_1 : write_date didn't change because the cr.now() is cached (Also the cache is already not correct)
+        # -> it is incorrect  here in the cache. old render is in the cache and it shouldn't be
+        # - rerender template X : it will rerender the same than the first render because the write_date didn't change but the value did
+        # then the template in cache won't be correct anymore (represent the old values)
+        self.tcache_pollution = {
```

Comment:
I would replace the concept of `pollution` with `stale`, it is more explicit IMHO